routes/index.js

```diff
@@ -1,7 +1,11 @@
 const express = require('express')
 const router = express.Router()
-const Model = require('../models/model')
+const User = require('../models/user')
 
+const users = require('../api/users-api')
 
+router.get('/', users.getUserByUsername, (req, res) => {
+    res.send(req.user)
+})
 
```

Comment:
Ovaj pattern nije dobar. Nema potrebe koristiti nešto nalik na middleware funkciju da bi se izmjenio req obejakt i vratio na frontend, a samo se želi dobaviti user. Dalje u responsu treba biti status number, i response se treba vratiti na frontend i u slučaju errora.

Reply:
Zašto je putanja samo '/' ako se geta user?

Reply:
Random primjer sa interneta koji je ispravan: const getStudents = async (req, res) => {
    try {
        const student= await Student.find();
        
        res.status(200).json(student);
    } catch(error) {
        res.status(404).json({message: error.message});
    }
}

Reply:
[Primjer tutorijala](https://blog.appsignal.com/2022/08/17/build-a-crud-app-with-nodejs-and-mongodb.html) gdje su fino razdvojeni , fino su razdvojeni servisi, kontroleri i putanje

Reply:
Više mi ima smisla da je putanja npr /user/:username